src/Operators/spectralelement.jl

```diff
@@ -631,6 +631,156 @@ Base.@propagate_inbounds function apply_operator(
+```math
+(Div_{split})_i = \\frac{1}{J_i} \\sum_j D_{ij} F_{ij}
+```
+where ``F_{ij}`` is the symmetric **two-point flux** between nodes ``i`` and ``j``:
+```math
+F_{ij} = \\frac{1}{2} \\left( U_i + U_j \\right) (\\psi_i + \\psi_j)
+```
+Here, ``U`` represents the volume-weighted contravariant flux component (``U^1 = J u^1``). In 2D/3D tensor-product grids, this 1D split operator is 
+applied sequentially along each dimension.
+
+# Properties
+1.  **Conservation:** The operator remains conservative.
+2.  **Consistency:** If `ψ` is spatially constant (e.g., `ψ = 1`), the operator degenerates to the standard divergence of `u` (mass continuity).
+3.  **Complexity:** Unlike standard spectral element divergence which uses tensor product factorization for ``O(N)`` cost, this operator requires explicit interaction between all node pairs on a 1D line, resulting in ``O(N^2)`` complexity per line (where ``N`` is the polynomial order).
+
+# References
+- Fisher, T. C., & Carpenter, M. H. (2013). High-order entropy stable finite difference schemes for nonlinear conservation laws: Finite domains. Journal of Computational Physics, 252, 518-557. [https://doi.org/10.1016/j.jcp.2013.06.014](https://doi.org/10.1016/j.jcp.2013.06.014)
+- Gassner, G. J. (2013). A skew-symmetric discontinuous Galerkin spectral element discretization and its relation to SBP-SAT finite difference methods. SIAM Journal on Scientific Computing, 35, A1233-A1253. [https://doi.org/10.1137/120890144](https://doi.org/10.1137/120890144)
+"""
+struct SplitDivergence{I} <: SpectralElementOperator{I} end
+SplitDivergence() = SplitDivergence{()}()
+SplitDivergence{()}(space) = SplitDivergence{operator_axes(space)}()
+
+operator_return_eltype(::SplitDivergence{I}, ::Type{S1}, ::Type{S2}) where {I, S1, S2} =
+    RecursiveApply.rmaptype(Geometry.divergence_result_type, S1)
+
+function apply_operator(op::SplitDivergence{(1,)}, space, slabidx, arg1, arg2)
+    FT = Spaces.undertype(space)
+    QS = Spaces.quadrature_style(space)
+    Nq = Quadratures.degrees_of_freedom(QS)
+    D = Quadratures.differentiation_matrix(FT, QS)
+    RT = operator_return_eltype(op, eltype(arg1), eltype(arg2))
+    out = IF{RT, Nq}(MArray, FT)
+    fill!(parent(out), zero(FT))
+
+    Ju1_slab = MArray{Tuple{Nq}, FT}(undef)
+    psi_slab = MArray{Tuple{Nq}, FT}(undef)
+
+    @inbounds for i in 1:Nq
+        ij = CartesianIndex((i,))
+        local_geometry = get_local_geometry(space, ij, slabidx)
+        u = get_node(space, arg1, ij, slabidx)
+        psi = get_node(space, arg2, ij, slabidx)
+        Ju1 = local_geometry.J * Geometry.contravariant1(u, local_geometry)
+        Ju1_slab[i] = Ju1
+        psi_slab[i] = psi
+    end
+
+    @inbounds for i in 1:Nq
+        val = zero(FT)
+        for j in 1:Nq
+            flux_ij = 0.5 * (Ju1_slab[i] + Ju1_slab[j]) * (psi_slab[i] + psi_slab[j])
```

Comment:
```suggestion
            flux_ij = (Ju1_slab[i] + Ju1_slab[j]) * (psi_slab[i] + psi_slab[j]) / 2
```
0.5 is a `Float64`, so this is a type instability when `FT = Float32`